src/Analyzer.JsonRuleEngine/JsonRuleResult.cs

```diff
@@ -30,5 +31,22 @@ internal class JsonRuleResult : IResult
         /// Gets the expression associated with this result
         /// </summary>
         internal Expression Expression { get; set; }
+
+        public bool Equals(JsonRuleResult other)
```

Comment:
All results have this identical code--create an abstract Result class so this functionality can just be shared and not duplicated in each implementation?

Reply:
This might be a little trickier than I thought since if I do this the simple way with the abstract class impl, a powershell and json rule engine result on the same line could be duped if I don't preserve the type comparison of each derived type (i.e I need to make sure the type of the of result factors in the equality as well when comparing). I am not sure yet if that matters though.

Reply:
A couple thoughts:
- If type does matter, you can likely just include that as another check in the implementation.  The concrete type should come through at runtime, even if implemented in the abstract base.
- It's also possible we don't need to keep separate implementations for JSON and PowerShell.  Maybe we can merge them.

Reply:
I don't think it matters since all evaluations will have the same types of results, and when results are compared across expressions (to dedupe), the rule type of the expression will also be a part of that comparison (assuming rule IDs are unique to each result type). Regardless, I will implement the type checking of the concrete type.

Reply:
I ended up merging all the results into one class. Let me know your thoughts, there's some JSON engine specific stuff that makes it a bit less clean.